src/Interpreters/MergeJoin.cpp

```diff
@@ -278,11 +278,11 @@ class MergeJoinCursor
     /// Expects !atEnd()
     size_t getEqualLength()
     {
-        size_t pos = impl.pos + 1;
+        size_t pos = impl.getPos() + 1;
         for (; pos < impl.rows; ++pos)
             if (!samePrev(pos))
                 break;
-        return pos - impl.pos;
+        return pos - impl.getPos();
```

Comment:
I started to change it in case when the cursor has permutation, but I realized that all logic with `MergeJoinEqualRange`, `getNextEqualRange` and their usage assume that the block in the cursor is sorted and it will be difficult to support permutation here. We explicitly initialize cursor without permutation here:
https://github.com/ClickHouse/ClickHouse/blob/5ae6c6dab9b06acc1707db44077346e7f0208bdd/src/Interpreters/MergeJoin.cpp#L182
So, I guess it's ok that we assume that the cursor doesn't have permutation in MergeJoin

Reply:
Ok. Let's then check in `MergeJoinCursor` constructor that permutation for `impl` is nullptr